packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx

```diff
@@ -0,0 +1,209 @@
+import {
+  BreadcrumbManager,
+  buildProductListing,
+  buildSearch,
+  Product,
+  ProductListing,
+  ProductTemplatesHelpers,
+  Search,
+  RegularFacetValue,
+} from '@coveo/headless/commerce';
+import {Component, Element, Prop, h, State, VNode} from '@stencil/core';
+import {getFieldValueCaption} from '../../../../utils/field-utils';
+import {InitializeBindings} from '../../../../utils/initialization-utils';
+import {titleToKebab} from '../../../../utils/utils';
+import {CommerceBindings} from '../../atomic-commerce-interface/atomic-commerce-interface';
+import {ProductContext} from '../product-template-decorators';
+
+/**
+ * The `atomic-product-multi-value-text` component renders the values of a multi-value string field.
+ * @part product-multi-value-text-list - The list of field values.
+ * @part product-multi-value-text-separator - The separator to display between each of the field values.
+ * @part product-multi-value-text-value - A field value.
+ * @part product-multi-value-text-value-more - A label indicating some values were omitted.
+ * @slot product-multi-value-text-value-* - A custom caption value that's specified for a given part of a multi-text field value. For example, if you want to use `Off-Campus Resident` as a caption value for `Off-campus apartment` in `Off-campus apartment;On-campus apartment`, you'd use `<span slot="product-multi-value-text-value-off-campus-apartment">Off-Campus Resident</span>`). The suffix of this slot corresponds with the field value, written in kebab case.
+ */
+@Component({
+  tag: 'atomic-product-multi-value-text',
+  styleUrl: 'atomic-product-multi-value-text.pcss',
+  shadow: true,
+})
+export class AtomicProductMultiValueText {
+  public breadcrumbManager!: BreadcrumbManager;
+  public searchOrListing!: Search | ProductListing;
+
+  @InitializeBindings() public bindings!: CommerceBindings;
+  @ProductContext() private product!: Product;
+
+  @Element() host!: HTMLElement;
+
+  @State() public error!: Error;
+
+  /**
+   * The field that the component should use.
+   * The component will try to find this field in the `Product.raw` object unless it finds it in the `Product` object first.
+   * Make sure this field is present in the `fieldsToInclude` property of the `atomic-commerce-interface` component.
+   */
+  @Prop({reflect: true}) public field!: string;
+
+  /**
+   * The maximum number of field values to display.
+   * If there are _n_ more values than the specified maximum, the last displayed value will be "_n_ more...".
```

Comment:
Actually I think the description accurately reflects the behavior we should expect, but the component's behavior is flawed.

If I set 'maxValuesToDisplay' to 2 and there are 3 values available in total, I would definitely expect 2 values to be displayed, and a string saying "1 more..." to be rendered.

Similarly, if `maxValuesToDisplay` is set to 1 and there are 3 values available in total, we should display 1 value and a string saying "2 more..."

Right now, the latter works, but not the former.

This is because of what we do in the `getNumberOfValuesToDisplay` function:

```typescript
  private getNumberOfValuesToDisplay(values: string[]) {
    if (values.length <= this.maxValuesToDisplay) {
      return values.length;
    }
    if (this.maxValuesToDisplay < 2) {
      return this.maxValuesToDisplay;
    }
    return Math.min(values.length - 2, this.maxValuesToDisplay);
  }
```

if we suppose `values.length === 3` and `this.maxValuesToDisplay == 2`, then the function will return `Math.min(values.length - 2, this.maxValuesToDisplay)`, i.e., 1. This is not what we want.